src/include/duckdb/storage/table/scan_state.hpp

```diff
@@ -246,9 +278,9 @@ class TableScanState {
 	~TableScanState();
 
 	//! The underlying table scan state
-	CollectionScanState table_state;
+	unique_ptr<CollectionScanState> table_state;
```

Comment:
I don't quite follow why all of these changes are required. This seems like it should only impact how we get the next row group - i.e. in `RowGroupCollection::NextParallelScan` we no longer get the next row group by calling `row_groups->GetNextSegment` - but instead get the next row group by consulting an ordered list of row groups.

Do we need to do this re-structuring?